### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by CZEMacLeod** [CZEMacLeod/MSBuild.SDK.SystemWeb#14] [Q&A] [answered]
I am considering setting up a docs folder and associated GitHub pages site to capture some of the stuff that has come up through issues and allow people to add their own documentation contributions through PR's.

I have added a couple of wiki pages with content already, and there is, of course, the information in the ReadMe.md files for the repo and the packages but I feel this is not enough - and doesn't easily let others contribute.

I have not done GitHub based documentation before, and wanted some input from the community about how to structure this and what to add. 
And I want to encourage everyone to contribute to the project and the documentation.

Please let me know your thoughts on this.

**Comment by fretje** [comment, marked as answer]
Yes, I think GitHub pages is the way to go for this. Maintaining wiki pages on a GitHub repo can be a PITA sometimes, as it works with a separate git repository. With GitHub pages you simply add a docs folder to this repository with markdown documents and activate it in the settings (couple of clicks), é voila...  :-)

That's as far as my experience goes with it though. I know it uses Jekyll, which has a lot of plugins and customizations, but I haven't gone beyond normal markdown and one of the default templates...

I'm also not the best with regards to writing or how to structure documentation in general, but I can definitely help fill in the blanks here and there once someone has set up some kind of structure ;-)

**Reply by CZEMacLeod** [reply]
I've pushed up some content to the [docs](https://github.com/CZEMacLeod/MSBuild.SDK.SystemWeb/tree/main/docs) folder and enabled [github pages](https://czemacleod.github.io/MSBuild.SDK.SystemWeb/).

I'm hoping that someone with a little more experience with this can help organise and get it working more smoothly.
I'm more than happy to accept any PRs to the ReadMe.md files in the main src tree, and any files in the docs folder.

Contributions to the main project and templates are also more than welcome. :)
